docs/architecture/adr-020-post-ownership-change.md

```diff
@@ -0,0 +1,83 @@
+# ADR 020: Post ownership transfer
+
+## Changelog
+- April 21th, 2023: First draft;
+
+## Status
+
+Proposed
+
+## Abstract
+
+This ADR introduces a new feature that enables users to transfer post ownership to another person.
+
+## Context
+
+Desmos is a social network protocol that allows users to create, share, and engage with content on a decentralized platform. As of now, Desmos does not provide a feature that allows users to transfer the ownership of their posts to other users. This has caused inconvenience for users who wish to transfer ownership of their posts, as they have to create new posts and lose the engagement history and feedback of the original post. Therefore, the introduction of the new feature that enables users to transfer post ownership to another person aims to address this issue and provide users with more control over their content. The proposed feature is expected to improve user experience and enhance the functionality of the Desmos protocol.
+
+## Decision
+
+We will add an __owner__ field to the post structure and implement a new handler that allows users to transfer post ownership to another person. The upcoming changes are as follows:
+
+### Type
+
+```proto
+message Post {
+  uint64 subspace_id = 1;
+  uint32 section_id = 2;
+  
+  ...skip
+
+  google.protobuf.Timestamp last_edited_date = 13;
+  
+  // Owner of the post
+  string owner = 14;
+}
+```
+
+### `Msg` Service
+
+```proto
+service Msg {
+    // ChangePostOwner allows users to transfer the ownership of a post to another person
+    rpc ChangePostOwner(MsgChangePostOwner) returns (MsgChangePostOwnerResponse);
```

Comment:
Can't we simply use the `MsgEditPost` message instead? Do we have any advantage in defining a new message type over using the existing one?

Reply:
Having a new message can follow SRP(Single responsibility principle), make sure `MsgEditPost` only modify the post content, which brings the benefit as follows:
1. User can avoid move ownership accidentally when editing a post, and vice versa
2. `MsgEditPost` related methods inside `desmos-bindings`, `desmjs` can stay away from breaking change
3. Tracker can easily track the ownership changes history of a post

As result, I decided to define a new message.